HWs/Test1.1/task3.c

```diff
@@ -0,0 +1,47 @@
+#include <stdio.h>
+#include <locale.h>
+
+#define sizeOfCodes 255
+
+int main()
+{
+    setlocale(LC_ALL, "RU-ru");
+
+    FILE *file = fopen("text.txt", "r");
+
+    if (file == NULL)
+    {
+        printf("Ошибка работы с файлом.\n");
+        return 0;
+    }
+
+    int frequencyDictionary[sizeOfCodes] = { 0 };
+
+    while(!feof(file))
+    {
+        ++frequencyDictionary[fgetc(file)];
```

Comment:
```suggestion
        int currentCharCode = fgetc(file);
        if (currentCharCode >= 0)
        {
            ++frequencyDictionary[currentCharCode];
        }
```
Здесь забавно получается. `fgetc` берет следующий символ из файла и переносит указатель на этот символ.
`feof` сообщит о том, что файл исчерпан, только когда указатель будет правее последнего символа в файле.
Вот и получается, что на последней итерации указатель в файле указывает на последний символ, `feof` запускает нас в тело цикла, `fgetc` возвращает следующий символ (символа нет, поэтому `-1`), и мы присваиваем по адресу `-1` в массив. То есть выходим за пределы массива.